src/lightcurvelynx/simulate.py

```diff
@@ -221,10 +333,100 @@ def simulate_lightcurves(
         # The number of observations is the total across all surveys.
         results_dict["nobs"][idx] = total_num_obs
 
-    # Create the nested frame.
+    # Create and return the nested frame.
     results = NestedFrame(data=results_dict, index=[i for i in range(num_samples)])
     nested_frame = pd.DataFrame(data=nested_dict, index=nested_index)
     results = results.add_nested(nested_frame, "lightcurve")
+    return results
+
+
+def simulate_lightcurves(
+    model,
+    num_samples,
+    obstable,
+    passbands,
+    *,
+    obstable_save_cols=None,
+    param_cols=None,
+    apply_obs_mask=False,
+    time_window_offset=None,
+    rng=None,
+    generate_citations=False,
+    pool=None,
+    batch_size=1_000,
+):
+    """Generate a number of simulations of the given model and information
+    from one or more surveys.
+
+    Parameters
+    ----------
+    model : BasePhysicalModel
+        The model to draw from. This may have its own parameters which
+        will be randomly sampled with each draw.
+    num_samples : int
+        The number of samples.
+    obstable : ObsTable or List of ObsTable
+        The ObsTable(s) from which to extract information for the samples.
+    passbands : PassbandGroup or List of PassbandGroup
+        The passbands to use for generating the bandfluxes.
+    apply_obs_mask: boolean
+        If True, apply obs_mask to filter interesting indices/times.
+    time_window_offset : tuple(float, float), optional
+        A tuple specifying the time window offset (before, after) t0 in days.
+        This is used to filter the observations to only those within the specified
+        time window (t0 - before, t0 + after). If None or the model does not have a
+        t0 specified, no time window is applied.
+    obstable_save_cols : list of str, optional
+        A list of ObsTable columns to be saved as part of the results. This is used
+        to save context information about how the light curves were generated. If the column
+        is missing from one of the ObsTables, a null value such as None or NaN is used.
+        If None, no additional columns are saved.
+    param_cols : list of str, optional
+        A list of the model's parameter columns to be saved as separate columns in
+        the results (instead of just the full dictionary of parameters). These
+        must be specified as strings in the node_name.param_name format.
+        If None, no additional columns are saved.
+    rng : numpy.random._generator.Generator, optional
+        A given numpy random number generator to use for this computation. If not
+        provided, the function uses the node's random number generator.
+    generate_citations : bool, optional
+        If True, generate citations for the simulation and output them.
+        Default is False.
+    pool : multiprocessing.Pool, optional
```

Comment:
`multiprocessing.Pool` is considered an older implementation, while `concurrent.futures.Executor` is a base class for a more modern Python standard library parallelism implementation. See the notes in the `multiprocessing` docs:
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.ThreadPool